From b5f4e75b747db09c882358fd85612966e840adb8 Mon Sep 17 00:00:00 2001
From: zkmike <zkmike@gmail.com>
Date: Tue, 16 Dec 2025 17:14:08 -0800
Subject: [PATCH 2/4] Remove OV5645 and added TC358743 power

---
 .../boot/dts/renesas/r9a09g056n44-evk.dts     | 106 +++++-------------
 1 file changed, 28 insertions(+), 78 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/r9a09g056n44-evk.dts b/arch/arm64/boot/dts/renesas/r9a09g056n44-evk.dts
index d35cb66138ea..3c0937464b05 100644
--- a/arch/arm64/boot/dts/renesas/r9a09g056n44-evk.dts
+++ b/arch/arm64/boot/dts/renesas/r9a09g056n44-evk.dts
@@ -245,28 +245,37 @@ clk_ext_camera: clk_ext_camera {
 		clock-frequency = <24000000>;
 	};
 
-	ov5645_vdddo_1v8: 1p8v {
-		compatible = "regulator-fixed";
-		regulator-name = "camera_vdddo";
-		regulator-min-microvolt = <1800000>;
-		regulator-max-microvolt = <1800000>;
-		regulator-always-on;
+
+	tc358743_dovdd_1v8: regulator-1v8 {
+	    compatible = "regulator-fixed";
+	    regulator-name = "tc358743_dovdd";
+	    regulator-min-microvolt = <1800000>;
+	    regulator-max-microvolt = <1800000>;
+	    regulator-always-on;
 	};
 
-	ov5645_vdda_2v8: 2p8v {
-		compatible = "regulator-fixed";
-		regulator-name = "camera_vdda";
-		regulator-min-microvolt = <2800000>;
-		regulator-max-microvolt = <2800000>;
-		regulator-always-on;
+	/* Add 3.3V regulator for analog (if not already available) */
+	tc358743_avdd_3v3: regulator-3v3 {
+	    compatible = "regulator-fixed";
+	    regulator-name = "tc358743_avdd";
+	    regulator-min-microvolt = <3300000>;
+	    regulator-max-microvolt = <3300000>;
+	    regulator-always-on;
+	};
+	/* Add a 1.2V regulator for TC358743 digital core */
+	tc358743_dvdd_1v2: regulator-1v2 {
+	    compatible = "regulator-fixed";
+	    regulator-name = "tc358743_dvdd";
+	    regulator-min-microvolt = <1200000>;
+	    regulator-max-microvolt = <1200000>;
+	    regulator-always-on;
 	};
 
-	ov5645_vddd_1v5: 1p5v {
-		compatible = "regulator-fixed";
-		regulator-name = "camera_vddd";
-		regulator-min-microvolt = <1500000>;
-		regulator-max-microvolt = <1500000>;
-		regulator-always-on;
+	/* Add 27MHz clock */
+	tc358743_clk: tc358743-clock {
+	    compatible = "fixed-clock";
+	    #clock-cells = <0>;
+	    clock-frequency = <27000000>;
 	};
 
 	hdmi-out {
@@ -822,27 +831,6 @@ &i2c0 {
 
 	status = "okay";
 
-	ov5645_csi20: ov5645_csi20@3c {
-		compatible = "ovti,ov5645";
-		reg = <0x3c>;
-		clock-names = "xclk";
-		clocks = <&clk_ext_camera>;
-		clock-frequency = <24000000>;
-		vdddo-supply = <&ov5645_vdddo_1v8>;
-		vdda-supply = <&ov5645_vdda_2v8>;
-		vddd-supply = <&ov5645_vddd_1v5>;
-		enable-gpios = <&pinctrl RZG2L_GPIO(11, 3) GPIO_ACTIVE_HIGH>;
-		/* Dummy pin for reset-gpios */
-		reset-gpios = <&pinctrl RZG2L_GPIO(6, 4) GPIO_ACTIVE_LOW>;
-
-		port {
-			ov5645_to_csi20: endpoint {
-				clock-lanes = <0>;
-				data-lanes = <1 2>;
-				remote-endpoint = <&csi20_in>;
-			};
-		};
-	};
 };
 
 &cru0 {
@@ -852,15 +840,6 @@ &cru0 {
 &csi20 {
 	status = "okay";
 
-	ports {
-		port {
-			csi20_in: endpoint {
-				clock-lanes = <0>;
-				data-lanes = <1 2>;
-				remote-endpoint = <&ov5645_to_csi20>;
-			};
-		};
-	};
 };
 
 &i2c1 {
@@ -869,27 +848,6 @@ &i2c1 {
 
 	status = "okay";
 
-	ov5645_csi21: ov5645_csi21@3c {
-		compatible = "ovti,ov5645";
-		reg = <0x3c>;
-		clock-names = "xclk";
-		clocks = <&clk_ext_camera>;
-		clock-frequency = <24000000>;
-		vdddo-supply = <&ov5645_vdddo_1v8>;
-		vdda-supply = <&ov5645_vdda_2v8>;
-		vddd-supply = <&ov5645_vddd_1v5>;
-		enable-gpios = <&pinctrl RZG2L_GPIO(11, 4) GPIO_ACTIVE_HIGH>;
-		/* Dummy pin for reset-gpios */
-		reset-gpios = <&pinctrl RZG2L_GPIO(6, 5) GPIO_ACTIVE_LOW>;
-
-		port {
-			ov5645_to_csi21: endpoint {
-				clock-lanes = <0>;
-				data-lanes = <1 2>;
-				remote-endpoint = <&csi21_in>;
-			};
-		};
-	};
 };
 
 &cru1 {
@@ -899,15 +857,7 @@ &cru1 {
 &csi21 {
 	status = "okay";
 
-	ports {
-		port {
-			csi21_in: endpoint {
-				clock-lanes = <0>;
-				data-lanes = <1 2>;
-				remote-endpoint = <&ov5645_to_csi21>;
-			};
-		};
-	};
+
 };
 #else
 &spi2 {
-- 
2.25.1

