From 4d0e590a4f7efe273fbd94eb08f1be9b7db3c8ba Mon Sep 17 00:00:00 2001
From: zkmike <zkmike@gmail.com>
Date: Thu, 23 Apr 2026 08:49:48 -0700
Subject: [PATCH 5/5] Add tc358743 camera CAM1 CN8

---
 .../dts/renesas/r9a09g056n48-rzv2n-evk.dts    | 42 +++++++++++++++++++
 1 file changed, 42 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r9a09g056n48-rzv2n-evk.dts b/arch/arm64/boot/dts/renesas/r9a09g056n48-rzv2n-evk.dts
index 99d84dd39..73ea0e1ec 100644
--- a/arch/arm64/boot/dts/renesas/r9a09g056n48-rzv2n-evk.dts
+++ b/arch/arm64/boot/dts/renesas/r9a09g056n48-rzv2n-evk.dts
@@ -518,6 +518,38 @@ &i2c1 {
 	pinctrl-names = "default";
 	clock-frequency = <400000>;
 	status = "okay";
+
+	tc358743_csi21: tc358743_csi21@f {
+
+	    /* TC358743 SPECIFIC - must match driver exactly */
+	    compatible = "toshiba,tc358743";
+		reg = <0x0f>;
+
+	    clock-names = "refclk";					// TC358743 Specific name
+	    dovdd-supply = <&tc358743_dovdd_1v8>;		// TC358743 Specific supply names
+	    avdd-supply = <&tc358743_avdd_3v3>;	// TC358743 Specific supply names  
+	    dvdd-supply = <&tc358743_dvdd_1v2>;		// TC358743 Specific supply names
+	    
+	    /* STANDARD properties (but TC358743-specific values) */
+	    clocks = <&tc358743_clk>;
+	    clock-frequency = <27000000>;  // HDMI typical frequency
+		
+		enable-gpios = <&pinctrl RZG2L_GPIO(6, 5) GPIO_ACTIVE_HIGH>;
+	    /*Enable the MIPI reset pin */
+		reset-gpios = <&pinctrl RZG2L_GPIO(10, 6) GPIO_ACTIVE_LOW>;
+	    
+	    port {
+	        tc358743_to_csi21: endpoint {
+	            /* STANDARD CSI-2 properties */
+	            clock-lanes = <0>;
+	            data-lanes = <1 2 3 4>;
+	            remote-endpoint = <&csi21_in>;
+	            
+	            /* TC358743 commonly uses this */
+	            link-frequencies = /bits/ 64 <297000000>;
+	        };
+	    };
+	};	
 };
 
 #if (I2C2_I3C_POEG_SEL == I2C2_SEL || I2C2_I3C_POEG_SEL == I3C_SEL)
@@ -1277,6 +1309,16 @@ &cru1 {
 
 &csi21 {
 	status = "okay";
+
+	ports {
+		port {
+			csi21_in: endpoint {
+				clock-lanes = <0>;
+				data-lanes = <1 2 3 4>;
+				remote-endpoint = <&tc358743_to_csi21>;
+			};
+		};
+	};
 };
 
 &cmtw0 {
-- 
2.25.1

